packages/react/src/components/DataTable/DataTable.tsx

```diff
@@ -294,6 +296,12 @@ class DataTable<RowType, ColTypes extends any[]> extends React.Component<
       })
     ).isRequired,
 
+    /**headerkey that we are sorting by */
+    initialSortColumn: PropTypes.string,
```

Comment:
If this is specifically the `key` on a header, we should name it as such

```suggestion
    /** Specify the key of a header to initially sort the table by */
    initialSortHeaderKey: PropTypes.string,
```